libs/libc/string/lib_strsignal.c

```diff
@@ -22,53 +22,24 @@
  * Included Files
  ****************************************************************************/
 
+#include <errno.h>
 #include <signal.h>
+#include <stdio.h>
 #include <string.h>
 
 /****************************************************************************
- * Private Data
+ * Pre-processor Definitions
  ****************************************************************************/
 
-/* We don't know what signals names will be assigned to which signals in
- * advance and we do not want to return a volatile value.  One solution is
- * this silly array of useless names:
- */
-
-static FAR const char *g_default_sigstr[32] =
-{
-  "Signal 0",
-  "Signal 1",
-  "Signal 2",
-  "Signal 3",
-  "Signal 4",
-  "Signal 5",
-  "Signal 6",
-  "Signal 7",
-  "Signal 8",
-  "Signal 9",
-  "Signal 10",
-  "Signal 11",
-  "Signal 12",
-  "Signal 13",
-  "Signal 14",
-  "Signal 15",
-  "Signal 16",
-  "Signal 17",
-  "Signal 18",
-  "Signal 19",
-  "Signal 20",
-  "Signal 21",
-  "Signal 22",
-  "Signal 23",
-  "Signal 24",
-  "Signal 25",
-  "Signal 26",
-  "Signal 27",
-  "Signal 28",
-  "Signal 29",
-  "Signal 30",
-  "Signal 31",
-};
+#ifdef CONFIG_LIBC_STRSIGNAL_SHORT
+#  define CASE_SIG_STR(sig, msg) \
+    case (sig): \
+      return (FAR char *)#sig
+#else
+#  define CASE_SIG_STR(sig, msg) \
+    case (sig): \
+      return (FAR char *)(msg)
+#endif
```

Comment:
Question: does macros like this work on old C89 pre-processor? This code is a good improvement, but we need to double check if we are not introducing incompatibilities.

Reply:
@acassis I think the macro should work correctly, but I cannot verify it at the moment. GCC can be made to build in C89 mode with the `-std=c89` flag, in case anyone wishes to verify it.

Reply:
There are no C89 pre-processor limitations for such macros that I'm aware of